controllers/iamrole_controller.go

```diff
@@ -127,7 +127,8 @@ func (r *IamroleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 func (r *IamroleReconciler) HandleReconcile(ctx context.Context, req ctrl.Request, iamRole *iammanagerv1alpha1.Iamrole) (ctrl.Result, error) {
```

Comment:
in general, this `HandleReconcile` method is just way too long, but it is doing a lot. Trying to refactor the error handling code to not be so repetitive (e.g. extracting a `logAndRecordReconcilationError` method) might shorten it and make it easier to follow.

Reply:
So I _completely_ agree. In this PR I would like to not make too many changes to he `HandleReconcile` method because I am concerned about trying to submit that upstream and have the Keiko team balk at a massive looking rewrite. Any significant objections to me punting on that _for now_?

Reply:
no